pennylane/transforms/split_non_commuting.py

```diff
@@ -384,17 +384,17 @@ def _split_ham_with_grouping(tape: qml.tape.QuantumScript):
 
 def _split_using_qwc_grouping(
     tape: qml.tape.QuantumScript,
-    single_term_obs_mps: dict[MeasurementProcess, tuple[list[int], list[float]]],
-    offsets: list[float],
+    single_term_obs_mps: dict[MeasurementProcess, tuple[list[int], list[Union[float, TensorLike]]]],
+    offsets: list[Union[float, TensorLike]],
```

Comment:
```suggestion
    offsets: list[TensorLike],
```
float is included in the definition of `TensorLike`.